core/plugin_api.php

```diff
@@ -674,24 +676,40 @@ function plugin_upgrade( MantisPlugin $p_plugin ) {
 
 		$t_target = $t_schema[$i][1][0];
 
-		if( $t_schema[$i][0] == 'InsertData' ) {
-			$t_sqlarray = array(
-				'INSERT INTO ' . $t_schema[$i][1][0] . $t_schema[$i][1][1],
-			);
-		} else if( $t_schema[$i][0] == 'UpdateSQL' ) {
-			$t_sqlarray = array(
-				'UPDATE ' . $t_schema[$i][1][0] . $t_schema[$i][1][1],
-			);
-			$t_target = $t_schema[$i][1];
-		} else if( $t_schema[$i][0] == 'UpdateFunction' ) {
-			$t_sqlarray = false;
-			if( isset( $t_schema[$i][2] ) ) {
-				$t_status = call_user_func( 'install_' . $t_schema[$i][1], $t_schema[$i][2] );
-			} else {
-				$t_status = call_user_func( 'install_' . $t_schema[$i][1] );
-			}
-		} else {
-			$t_sqlarray = call_user_func_array( array( $t_dict, $t_schema[$i][0] ), $t_schema[$i][1] );
+		switch( $t_schema[$i][0] ) {
+			case 'InsertData':
+				$t_sqlarray = array(
+					'INSERT INTO ' . $t_schema[$i][1][0] . $t_schema[$i][1][1],
+				);
+				break;
+
+			case 'UpdateSQL':
+				$t_sqlarray = array(
+					'UPDATE ' . $t_schema[$i][1][0] . $t_schema[$i][1][1],
+				);
+				$t_target = $t_schema[$i][1];
+				break;
+
+			case 'UpdateFunction':
+				$t_sqlarray = false;
+				if( isset( $t_schema[$i][2] ) ) {
+					$t_status = call_user_func( 'install_' . $t_schema[$i][1], $t_schema[$i][2] );
+				} else {
+					$t_status = call_user_func( 'install_' . $t_schema[$i][1] );
+				}
+				break;
+
+			case null:
+				# No-op upgrade step
+				$t_sqlarray = false;
+				$t_status = 2;
+				break;
+
+			default:
```

Comment:
Ideally, we would error on default and have all the cases specifically handled.  That is not part of the scope of this consistency fix, but would be worth considering.

Reply:
That's a good point, as unlike the MantisBT installer, we don't control the contents of the plugin schemas

Reply:
Well, on second thoughts that would force us to list every possible ADOdb dictionary method... 

I think leaving _default_ is OK, especially considering that 
- we'd get a PHP warning from *call_user_func()*, which would then return a *null* status if given an invalid function name, 
- if given an existing function that does not return valid SQL statement code, ExecuteSQLArray() would fail

In both cases,  an ERROR_PLUGIN_UPGRADE_FAILED would be triggered.